scripts/mypy.sh

```diff
@@ -1,7 +1,4 @@
 #!/usr/bin/env bash
 # Run type checking over the python code.
 
-set -e
-echo 'Starting mypy checks'
-mypy allennlp --ignore-missing-imports
-echo -e "mypy checks passed\n"
+./scripts/verify.py --checks mypy
```

Comment:
This will change the behavior of CI, right?  Did you check that TeamCity still gives us the same output if these checks fail?

Reply:
Ah shoot--you're right, I need the `set -e`.

I'm not sure what you mean by "add some unnecessary output".  Would you clarify?  What output would be added?

Reply:
I'm not certain this will happen, but it sure looks like you'll get an extra stack trace from the python code that isn't really necessary (the `CalledProcessError` from your call to `subprocess.run()` _should_ print a stack trace to the console).  It'd be nice to run this through CI with a failing test case or something just to see what happens.

Reply:
Nevermind--the exit code will still be the same.  I don't need `set -e`.

Reply:
Ah, good point.  I'll test and verify.  I don't think this will be a problem, but I understand your concern.

Reply:
@matt-gardner you're right.  I added some code to squelch the extra exception.

```
Traceback (most recent call last):
  File "./scripts/verify.py", line 52, in <module>
    main(run_checks)
  File "./scripts/verify.py", line 25, in main
    run("mypy allennlp --ignore-missing-imports", shell=True, check=True)
  File "/Users/michaels/miniconda3/envs/allennlp/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command 'mypy allennlp --ignore-missing-imports' returned non-zero exit status 1.
```